src/plugins/saml-auth/actions.js

```diff
@@ -0,0 +1,173 @@
+    case !!parseError:
+      return [data, JSON.stringify(data)]
+  }
+  return [data]
+}
+
+const authId = "SamlAuth"
+export const authenticateWithSamlToken = (schema, samlToken) => async ( { fn, samlAuthActions, authActions, errActions } ) => {
+  samlAuthActions.setSamlAuthState(SAML_AUTH_STATE_LOGGING_IN)
+
+  const method = schema.get("authMethod") || "post"
+  const authQuery = schema.get("authQuery")
+  const query = authQuery && authQuery.toJS()
+
+  const fetchUrl = appendQuery(schema.get("authTokenUrl"), query)
+
+  // 1. decode jwt
+  const [decoded, decodeErr] = decodeJWT(samlToken)
+  // guard error
+  if(decodeErr) {
+    errActions.newAuthErr({
+      authId,
+      level: "",
+      source: "auth",
+      message: `JWT Error: ${decodeErr.message}`
+    })
+    samlAuthActions.setSamlAuthState(SAML_AUTH_STATE_FAILED)
+    throw decodeErr
+  }
+
+  // 2. send request
+  const authBody = schema.get("authBody") || new Map()
+  const extraBody = authBody.toJS()
+  const body = JSON.stringify({ email: decoded.sub, saml_token: samlToken, ...extraBody })
+
+  const headers = {
+    "Accept":"application/json, text/plain, */*",
+    "Content-Type": "application/json"
+  }
+
+  let [response, fetchErr] = await (async function request() {
+    try {
+      return [await fn.fetch({
+        url: fetchUrl,
+        method: method || "post",
+        headers,
+        ...(method === "get" || method === "head" ? {} : {body})
+      }), undefined]
+    } catch(e) {
+      return [undefined, e]
+    }
+  })()
+  // guard error
+  if(fetchErr) {
+    const unauthorizedError = new Error(fetchErr)
+    unauthorizedError.message = fetchErr.response
+      ? fetchErr.response.body.message
+      : fetchErr.message
+
+    errActions.newAuthErr({
+      authId,
+      level: "",
+      source: "auth",
+      message: `Error logging in. ${unauthorizedError.message}`
+    })
+    samlAuthActions.setSamlAuthState(SAML_AUTH_STATE_FAILED)
+    throw unauthorizedError
+  }
+
+
+  // 3. parse response
+  const [data, parseErrorMessage] = parseLoginResponse(response)
+  if(parseErrorMessage) {
+    errActions.newAuthErr({
+      authId,
+      level: "",
+      source: "auth",
+      message: `Response Error: ${parseErrorMessage}`
+    })
+    samlAuthActions.setSamlAuthState(SAML_AUTH_STATE_FAILED)
+    throw new Error(parseErrorMessage)
+  }
+
+  // 4. set auth
+  authActions.authorize({
+    [authId]: {
+      name: authId,
+      email: decoded.sub,
+      token: data.token,
+      schema
+    }
+  })
+  samlAuthActions.saveSamlAuthEmail(decoded.sub)
+  samlAuthActions.setSamlAuthState(SAML_AUTH_STATE_LOGGED_IN)
+}
+
+export const loginSaml = (name, schema) => async ( { authActions } ) => {
+  const loginUrl = schema.get("loginUrl")
+  const loginQuery = schema.get("loginQuery")
+
+  if(!loginUrl) authActions.newAuthErr({
+    authId: name,
+    level: "error",
+    source: "auth",
+    message: "Swagger Config: Login URL not found"
+  })
```

Comment:
no need to handle all these unnecessary configs. It is not likely to change.
We only make things configurable if necessary, e.g. diff env.
`KISS` principle pls. Same for logout

Reply:
Need to fix [this](https://github.com/dsaidgovsg/lswagger-ui/pull/58#discussion_r1243793447) 1st then only can remove the check safely